### ISSUE TITLE
aerohive access points don't report clients

### ISSUE THREAD

**Issue by inphobia** [netdisco/snmp-info#325]
aerohive access points do not show their connected clients.

reports -> access points -> access point client count returns "no matching records"
navigating to a device -> ports & select show connected nodes & devices returns no clients.

i am trying to figure out what info is missing to correctly map the clients to the access points, but was not  succesful.

## Expected Behavior

like on wired ports, aerohive wireless interfaces should show connected clients.

## Current Behavior

no clients are show.

up until recently no information about clients was returned for aerohive devices. other features like ssid <-> port mapping, channel power have always worked fine

with commit https://github.com/netdisco/netdisco/commit/1dc3e70e37e0eadc1d3b51953d7598200b0de2a3 the node_wireless table is now filling up. with the exception of columns maxrate, sigrate & sigqual everything is filled in.

mac|ssid|uptime|maxrate|txrate|sigstrength|sigqual|rxpkt|txpkt|rxbyte|txbyte|time_last
---|----|------|-------|------|-----------|-------|-----|-----|------|------|---------
34:f3:9a:ed:55:71|Aquafin|2741|NULL|NULL|9|NULL|2894|1391|618862|672304|2019-01-31 16:41:02.975564

node_ip also contains wireless clients now.

mac|ip|active|time_first|time_last
---|--|------|----------|---------
34:f3:9a:ed:55:71|10.50.113.155|False|2019-01-16 12:20:41.391823|2019-01-16 16:20:25.624701
34:f3:9a:ed:55:71|10.50.114.47|True|2019-01-31 16:20:40.557098|2019-01-31 16:20:43.360859
34:f3:9a:ed:55:71|10.50.114.63|False|2019-01-30 16:20:42.50861|2019-01-30 16:20:45.548738
34:f3:9a:ed:55:71|10.50.114.148|False|2019-01-17 12:20:13.177149|2019-01-18 16:20:52.754059
34:f3:9a:ed:55:71|10.50.114.165|False|2019-01-22 12:20:33.166803|2019-01-22 12:20:32.739265

finally, in node they are also known, even on the correct vlan, but mapped to the switchport where the access point is connected to.

mac|switch|port|vlan|active|oui|time_first|time_recent|time_last
---|------|----|----|------|---|----------|-----------|--------
34:f3:9a:ed:55:71|10.40.254.12|GigabitEthernet1/0/45|113|True|34:f3:9a|2019-01-16 12:41:58.195312|2019-01-16 12:41:58.195312|2019-01-16 16:40:27.81937

**UPDATE**
when looking at the above table i just noticed time_last was 15days ago. so so wireless clients used to be matched to the switchport the access point was connected to, now they're just somewhere in limbo.
**END UPDATE**

also, all functions defined in snmp::info::layer2::airohive show correct data
(this is not the access point the client i used in the above example was connected to,since at the time of testing he is not active)

```
netdisco-do show -d hiveap215 -e i_ssidlist
[24061] 2019-01-31 19:40:10  info App::Netdisco version 2.040003 loaded.
[24061] 2019-01-31 19:40:10  info show: [10.40.252.36]/i_ssidlist started at Thu Jan 31 20:40:10 2019
\ {
    14   "AquafinBYOD",
    17   "AquafinSecure",
    18   "Aquafin",
    19   "AquafinGuest",
    20   "Aquafin",
    21   "AquafinGuest"
}
[24061] 2019-01-31 19:40:10  info show: finished at Thu Jan 31 20:40:10 2019
[24061] 2019-01-31 19:40:10  info show: status done: Showed i_ssidlist response from 10.40.252.36

netdisco-do show -d hiveap215 -e i_ssidmac
[24079] 2019-01-31 19:40:18  info App::Netdisco version 2.040003 loaded.
[24079] 2019-01-31 19:40:18  info show: [10.40.252.36]/i_ssidmac started at Thu Jan 31 20:40:18 2019
\ {
    14   "c4:13:e2:44:70:54",
    17   "c4:13:e2:44:70:55",
    18   "c4:13:e2:44:70:56",
    19   "c4:13:e2:44:70:57",
    20   "c4:13:e2:44:70:65",
    21   "c4:13:e2:44:70:66"
}
[24079] 2019-01-31 19:40:19  info show: finished at Thu Jan 31 20:40:19 2019
[24079] 2019-01-31 19:40:19  info show: status done: Showed i_ssidmac response from 10.40.252.36

netdisco-do show -d hiveap215 -e i_80211channel
[24097] 2019-01-31 19:40:33  info App::Netdisco version 2.040003 loaded.
[24097] 2019-01-31 19:40:33  info show: [10.40.252.36]/i_80211channel started at Thu Jan 31 20:40:33 2019
\ {
    7   6,
    8   56
}
[24097] 2019-01-31 19:40:34  info show: finished at Thu Jan 31 20:40:34 2019
[24097] 2019-01-31 19:40:34  info show: status done: Showed i_80211channel response from 10.40.252.36

netdisco-do show -d hiveap215 -e dot11_cur_tx_pwr_mw
[24108] 2019-01-31 19:40:45  info App::Netdisco version 2.040003 loaded.
[24108] 2019-01-31 19:40:45  info show: [10.40.252.36]/dot11_cur_tx_pwr_mw started at Thu Jan 31 20:40:45 2019
\ {
    7   16,
    8   18
}
[24108] 2019-01-31 19:40:46  info show: finished at Thu Jan 31 20:40:46 2019
[24108] 2019-01-31 19:40:46  info show: status done: Showed dot11_cur_tx_pwr_mw response from 10.40.252.36

netdisco-do show -d hiveap215 -e cd11_port
[24127] 2019-01-31 19:41:01  info App::Netdisco version 2.040003 loaded.
[24127] 2019-01-31 19:41:01  info show: [10.40.252.36]/cd11_port started at Thu Jan 31 20:41:01 2019
\ {
    17.6.80.130.213.156.162.49   "wifi0.2"
}
[24127] 2019-01-31 19:41:02  info show: finished at Thu Jan 31 20:41:02 2019
[24127] 2019-01-31 19:41:02  info show: status done: Showed cd11_port response from 10.40.252.36

netdisco-do show -d hiveap215 -e cd11_mac
[24135] 2019-01-31 19:41:10  info App::Netdisco version 2.040003 loaded.
[24135] 2019-01-31 19:41:10  info show: [10.40.252.36]/cd11_mac started at Thu Jan 31 20:41:10 2019
\ {
    17.6.80.130.213.156.162.49   "50:82:d5:9c:a2:31"
}
[24135] 2019-01-31 19:41:11  info show: finished at Thu Jan 31 20:41:11 2019
[24135] 2019-01-31 19:41:11  info show: status done: Showed cd11_mac response from 10.40.252.36

netdisco-do show -d hiveap215 -e bp_index
[24142] 2019-01-31 19:41:18  info App::Netdisco version 2.040003 loaded.
[24142] 2019-01-31 19:41:18  info show: [10.40.252.36]/bp_index started at Thu Jan 31 20:41:18 2019
\ {
    1    1,
    2    2,
    3    3,
    5    5,
    7    7,
    8    8,
    11   11,
    14   14,
    16   16,
    17   17,
    18   18,
    19   19,
    20   20,
    21   21
}
[24142] 2019-01-31 19:41:18  info show: finished at Thu Jan 31 20:41:18 2019
[24142] 2019-01-31 19:41:18  info show: status done: Showed bp_index response from 10.40.252.36

netdisco-do show -d hiveap215 -e qb_fw_port
[24148] 2019-01-31 19:41:25  info App::Netdisco version 2.040003 loaded.
[24148] 2019-01-31 19:41:25  info show: [10.40.252.36]/qb_fw_port started at Thu Jan 31 20:41:25 2019
\ {
    17.6.80.130.213.156.162.49   17
}
[24148] 2019-01-31 19:41:26  info show: finished at Thu Jan 31 20:41:26 2019
[24148] 2019-01-31 19:41:26  info show: status done: Showed qb_fw_port response from 10.40.252.36

netdisco-do show -d hiveap215 -e qb_fw_mac
[24153] 2019-01-31 19:41:34  info App::Netdisco version 2.040003 loaded.
[24153] 2019-01-31 19:41:34  info show: [10.40.252.36]/qb_fw_mac started at Thu Jan 31 20:41:34 2019
\ {
    17.6.80.130.213.156.162.49   "50:82:d5:9c:a2:31"
}
[24153] 2019-01-31 19:41:34  info show: finished at Thu Jan 31 20:41:34 2019
[24153] 2019-01-31 19:41:34  info show: status done: Showed qb_fw_mac response from 10.40.252.36

netdisco-do show -d hiveap215 -e qb_fw_vlan
[24157] 2019-01-31 19:41:40  info App::Netdisco version 2.040003 loaded.
[24157] 2019-01-31 19:41:40  info show: [10.40.252.36]/qb_fw_vlan started at Thu Jan 31 20:41:40 2019
\ {
    17.6.80.130.213.156.162.49   460
}
[24157] 2019-01-31 19:41:40  info show: finished at Thu Jan 31 20:41:40 2019
[24157] 2019-01-31 19:41:40  info show: status done: Showed qb_fw_vlan response from 10.40.252.36

netdisco-do show -d hiveap215 -e at_paddr
[24162] 2019-01-31 19:41:50  info App::Netdisco version 2.040003 loaded.
[24162] 2019-01-31 19:41:50  info show: [10.40.252.36]/at_paddr started at Thu Jan 31 20:41:50 2019
\ {
    17.6.80.130.213.156.162.49   "50:82:d5:9c:a2:31"
}
[24162] 2019-01-31 19:41:50  info show: finished at Thu Jan 31 20:41:50 2019
[24162] 2019-01-31 19:41:50  info show: status done: Showed at_paddr response from 10.40.252.36

netdisco-do show -d hiveap215 -e at_netaddr
[24168] 2019-01-31 19:41:57  info App::Netdisco version 2.040003 loaded.
[24168] 2019-01-31 19:41:57  info show: [10.40.252.36]/at_netaddr started at Thu Jan 31 20:41:57 2019
\ {
    17.6.80.130.213.156.162.49   "10.46.4.27"
}
[24168] 2019-01-31 19:41:57  info show: finished at Thu Jan 31 20:41:57 2019
[24168] 2019-01-31 19:41:57  info show: status done: Showed at_netaddr response from 10.40.252.36

```

actual netdisco display:
![image](https://user-images.githubusercontent.com/39005454/52080960-54890280-2599-11e9-82cf-918aebc9c3af.png)

## Possible Solution

the only guess i have at the moment is that no mapping is since no vlans are known on the airohive access points, which is strange since qb_fw_vlan() returns correct info.

## Steps to Reproduce (for bugs)

1. check web interface for connected clients on airohive access points

## Your Environment

* Netdisco version used: 2.40.3
* SNMP::Info version used: 3:64

**Comment by ollyg** [comment]
OK a little further digging. @JeroenvIS gave me a good pointer with IF-MIB versus RFC1213 and you can see from these two threads that loading RFC1213 only gives headaches:

* https://sourceforge.net/p/net-snmp/mailman/message/16013436/
* https://sourceforge.net/p/net-snmp/mailman/message/27684265/

Quite strong wording that we should NOT be loading RFC1213, but we are!

https://github.com/netdisco/snmp-info/blob/c4ce6a206285a59aa846cae9810ce5d434f1acec/lib/SNMP/Info.pm#L3364

This MIB is only loaded for the legacy ipRoute table:

https://github.com/netdisco/snmp-info/blob/c4ce6a206285a59aa846cae9810ce5d434f1acec/lib/SNMP/Info.pm#L3329

I ran some A/B tests with DebugSNMP enabled and it showed the bug not appearing when not loading RFC1213. Now, there is a simple fix and a fancy but awkward fix.

The simple fix is to sort the MIB names before we load them:

https://github.com/netdisco/snmp-info/blob/c4ce6a206285a59aa846cae9810ce5d434f1acec/lib/SNMP/Info.pm#L3953

Replace with `sort keys %$mibs`. RFC1213 comes after IF-MIB, so all good.

The fancier way would be to remove RFC1213 altogether and either completely drop support for ipRoute table, or move it to a custom class which has to be loaded on demand, or perhaps do some lazy-load of the MIB when the user calls anything `ipr_*` related.

**Comment by ollyg** [comment]
Per commit c73bc0ca2c43017382af3b14b0010f67f3fd8901, I have suppressed RFC1213-MIB from loading automatically and done it lazily if someone calls one of the `ipr_*` methods or prefixes their method (func call) with RFC1213-MIB.

This seems a reasonable approach for what is a legacy MIB and what the net-snmp list advise since 2005 that we should not load.

However @jeneric did write a bunch of tests explicitly to handle the variable loading order of IF-MIB and RFC1213-MIB (due to using Hash keys for `%mibs` to load) so I worry that his Great Mind is aware of some good reason for still loading RFC1213.

Anyway, shall we call this one closed?

(I have recommended to @inphobia that we leave in all the workarounds and tests they added for Aerohive, just for regression protection)
